packages/table/src/types/BulkRecordAction.ts

```diff
@@ -1,6 +1,14 @@
+import { DropdownItemProps } from 'reactstrap';
+import { IdType, Row } from './ReactTable';
 import { TableAction } from './TableAction';
 
-export interface BulkRecordAction<T> extends TableAction {
-  onClick?: (records?: T[]) => void;
-  isVisible?: (records?: T[]) => boolean;
+export interface BulkRecordAction<T extends IdType> extends TableAction {
+  onClick?: (records?: T[], rows?: Row<T>[]) => void;
+  isVisible?: (records?: T[], rows?: Row<T>[]) => boolean;
```

Comment:
Adding a second optional prop here to make this semi-backwards compatible. Not sure why we didn't just pass the row in the first place, but you know.... We have a scenario where we are utilizing the useExpanded hook on the react table an in our implementation we need to know the state of the isExpanded prop on the record. This also will allow access to other Row properties.